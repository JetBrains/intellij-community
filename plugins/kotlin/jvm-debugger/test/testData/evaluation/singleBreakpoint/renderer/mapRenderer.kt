package mapRenderer

private class MyMap : AbstractMap<String, String>() {
    private val implementationDetail = mapOf("a" to "aaaa", "b" to "bbbb")

    override val entries: Set<Map.Entry<String, String>>
        get() = implementationDetail.entries
}

fun main() {
    val emptyMap = emptyMap<String, String>()
    val nonEmptyMap = mapOf<String, String>(
        "k1" to "v1",
        "k2" to "v2"
    )

    val builtEmptyMap = buildMap<String, String> {}
    val builtNonEmptyMap = buildMap<String, String> {
        put("bk1", "bv1")
        put("bk2", "bv2")
    }

    val myMap = MyMap()

    //Breakpoint!
    run {}
}

// PRINT_FRAME