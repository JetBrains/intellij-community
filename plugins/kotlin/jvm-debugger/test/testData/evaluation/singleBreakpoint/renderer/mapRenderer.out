LineBreakpoint created at mapRenderer.kt:26
Run Java
Connected to the target VM
mapRenderer.kt:26
KotlinStackFrame (mapRenderer.kt:26)
    JavaValue[local] emptyMap: java.util.Map =  size = 0 (mapRenderer.kt:11)
    JavaValue[local] nonEmptyMap: java.util.Map =  size = 2 (mapRenderer.kt:12)
        JavaValue[element] 0 = ... -> ...
            JavaValue key: java.lang.String = k1
            JavaValue value: java.lang.String = v1
        JavaValue[element] 1 = ... -> ...
            JavaValue key: java.lang.String = k2
            JavaValue value: java.lang.String = v2
    JavaValue[local] builtEmptyMap: java.util.Map =  size = 0 (mapRenderer.kt:17)
    JavaValue[local] builtNonEmptyMap: java.util.Map =  size = 2 (mapRenderer.kt:18)
        JavaValue[element] 0 = ... -> ...
            JavaValue key: java.lang.String = bk1
            JavaValue value: java.lang.String = bv1
        JavaValue[element] 1 = ... -> ...
            JavaValue key: java.lang.String = bk2
            JavaValue value: java.lang.String = bv2
    JavaValue[local] myMap: mapRenderer.MyMap =  size = 2 (mapRenderer.kt:23)
        JavaValue[element] 0 = ... -> ...
            JavaValue key: java.lang.String = a
            JavaValue value: java.lang.String = aaaa
        JavaValue[element] 1 = ... -> ...
            JavaValue key: java.lang.String = b
            JavaValue value: java.lang.String = bbbb
mapRenderer.kt:26
Disconnected from the target VM

Process finished with exit code 0
